ERCS/erc-7546.md

```diff
@@ -0,0 +1,199 @@
+```
+The implementation contract MUST implement the function selector to be registered as a key in the mapping (section 3.1). Therefore, it is RECOMMENDED to verify this during registration.
+
+### 2. Proxy Contract
+This contract checks the Dictionary for the associated implementation address based on its function selector and delegatecalls to it.
+
+#### 2.1 Storage & Events
+The Proxy MUST reference the Dictionary address while ensuring a minimal chance of storage clashes. This is because the Dictionary's registered implementations will use this shared storage.
+
+Changes to the Dictionary address SHOULD be noticed:
+```solidity
+event DictionaryUpgraded(address indexed dictionary);
+```
+
+While it is possible to store the Dictionary address in the code area (e.g., using Solidity's immutable or constant), it SHOULD be designed with caution, considering the possibility that if the Dictionary's admin is not the same as the Proxy's admin, the ability to manipulate the implementation could be permanently lost.
+
+#### 2.2 Functions
+All calls to the Proxy MUST be forwarded to the implementation address registered in the Dictionary through the fallback function. Furthermore, as there is a possibility of collisions with function selectors registered in the Dictionary, Proxy SHOULD NOT have external functions.
+
+##### 2.2.1 `constructor`
+It is RECOMMENDED to initialize contracts, other than setting up the Dictionary address, through the Dictionary as well.
+
+##### 2.2.2 `fallback`
+It is RECOMMENDED to utilize well-established and widely-used logic from libraries like Zeppelin or Safe for the logic used in the fallback function. The implementation address to which delegatecall MUST be the returned value queried to the Dictionary (stored in the slot specified in section 2.1) using the `getImplementation(bytes4 functionSelector)` function defined in section 1.2.1.
+
+
+### 3. Interfaces
+The implementation contract MUST have the same function selector registered in the Dictionary. If not, the Proxy's delegatecall will fail.
+
+For transparency, it's RECOMMENDED for the Proxy to support interface queries from both on-chain and off-chain entities.
+
+#### 3.1 For on-chain support:
+To allow external contracts to ascertain the Proxy's behavior, the Dictionary and implementation contracts are RECOMMENDED to provide certain functions and interfaces.
+
+##### 3.1.1 Dictionary:
+###### 3.1.1.1 `supportsInterface`
+`supportsInterface(bytes4 interfaceID)` as defined in [ERC-165](./eip-165.md).
+###### 3.1.1.2 `supportsInterfaces`
+Implement `supportsInterfaces()` to return a list of registered interfaceIDs.
+```solidity
+function supportsInterfaces() public view returns (bytes4[] memory);
+```
+
+##### 3.1.2 Implementation Contracts:
+Since the Proxy MUST forward all calls to the Dictionary, the implementation for ERC-165 is RECOMMENDED to be registered in the Dictionary's mapping instead.
```

Comment:
Can you clarify what kind of forward do you refer to here? CALL/ STATICCALL / DELEGATECALL ,etc.

Reply:
https://github.com/ethereum/ERCs/pull/39/commits/5575a3e1582394d11a2b083096007b37bb856fc7